### PR TITLE
implement /submitStudentSurvey

### PR DESCRIPTION
**Pull request by antonlaukemper**

### CODE REVIEW

**Review by rauwuckl — APPROVED** [review]

File: Backend/scala-backend/app/controllers/SurveyController.scala [hunk 1]
```diff
@@ -18,17 +18,79 @@ import scala.concurrent.Future
...
+                                    val sourceId: Future[Option[Int]] = studentModel.createStudent(ego, id)
+                                    sourceId.flatMap(sId => sId match {
+                                        case Some(sId) => {
+                                            // enter each alter and store the id of the student objects
+                                            val targetIds: Seq[Future[Option[Int]]] = alters.map(alter => studentModel.createStudent(alter, id))
+                                            // create a new relationship with the alter as target and ego as source
+                                            targetIds.foreach(targetId => {
+                                                targetId.map(tId => {
+                                                    val relationship = RelationshipCC(classId=id, sourceId=sId, targetId=tId.get) //tId is an option
+                                                    relModel.createRelationship(relationship)
```
Comment (rauwuckl) [inline Backend/scala-backend/app/controllers/SurveyController.scala:66]: doulbe  check this is actually executed

File: Backend/scala-backend/app/models/StudentModel.scala [hunk 2]
```diff
@@ -7,8 +7,46 @@ import scala.concurrent.Future
...
-  ): Future[Boolean] = ???
+  // returns the id of the student inserted
+  def createStudent(studentCC: StudentCC, classId: Int): Future[Option[Int]] = {
+    
+    val resultRows: Future[Seq[StudentRow]] = db.run(Student.filter(_.hashedname === studentCC.hashedName).result)
+    val userExists: Future[Boolean] = resultRows.map(rows => rows.length != 0)
+    val studentId: Future[Int] = resultRows.map(rows => rows.head.id)
+    val alreadySelfReported: Future[Boolean]= resultRows.map(rows => rows.head.selfreported)
+    
+    userExists.flatMap(exists => {
```
Comment (rauwuckl) [inline Backend/scala-backend/app/models/StudentModel.scala:18]: call flatMap on resultRows. take out userExists, studentId there as direct values

File: Backend/scala-backend/app/models/StudentModel.scala [hunk 3]
```diff
@@ -7,8 +7,46 @@ import scala.concurrent.Future
...
+      // if the student already exists we need to differentiate between two cases
+      if (exists) {
+        alreadySelfReported.flatMap(sr => {
+          // if the existing student was already self reported and the current one is too
+          // we return None to signal a wrong input
+          if(sr && studentCC.selfReported) Future.successful(None)//return None because student already exists
+          // if the new entry is self reported but the old one is not, we need to update the value
+          else if (!sr && studentCC.selfReported) {
+            println("User already existed, updating selfReport status")
```
Comment (rauwuckl) [inline Backend/scala-backend/app/models/StudentModel.scala:27]: remove print

**Review by rauwuckl — CHANGES_REQUESTED** [review]

File: Backend/scala-backend/app/controllers/SurveyController.scala (hunk 1 above)
Comment (antonlaukemper) [reply Backend/scala-backend/app/controllers/SurveyController.scala:66]: it is 👍

**Pull request merged**